Refactor syntax unicode stack

livecodepanos opened the pull request:
Updated fields and methods of the Stack Object

livecodepanos committed:
Changed signatures of MCS_* methods

livecodepanos committed:
I have changed the signatures and the implementation of the methods. It compiles, but I get an exception in server.exe

livecodesebastien committed:
Implementation of MCAutoStringAs<string type> classes

livecodepanos committed:
Changed some signatures of MCSystem::* functions in srvmac.cpp and srvposix.cpp (these are not compiled). Also improved the code in srvwindows.cpp, and changed the variable names to more readable ones. Mark fixed the exception error

livecodepanos committed:
Did some more improvements in code

livecodesebastien committed:
MCS_ filesystem functions implemented

livecodepanos committed:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode

livecodepanos committed:
Corrections after Mark's feedback

runrevmark committed:
Added MCSTR() to create a constant MCStringRef from a cstring constant.

livecodepanos committed:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode

livecodepanos committed:
Modifications after Mark's feedback#2

livecodesebastien committed:
Merge with Panos branch and refactoring unicode of MCS_ functions and MCSystem functions

livecodesebastien committed:
Refactoring of MCSystemFileHandle now compiles

livecodesebastien committed:
Merge with Panos branch

runrevmark committed:
Added MCStringCreateWithCStringAndRelease.

livecodepanos committed:
Modifications after Mark's feedback #3

livecodesebastien committed:
First MCS_* functions refactoring step finished - engine is broken for now

livecodesebastien committed:
Correction of startup crash. The engine is still not properly working

livecodesebastien committed:
Creation of MCDataRef from bytes fixed - did not copy the data before...
MCMacDesktop::PathToNative and PathFromNative updated

livecodesebastien committed:
Merge with Panos' branch

livecodepanos committed:
Some signature updates in the MacOS related files (apart from osxfiles.cpp and osxspec.cpp, since they will be removed)

livecodepanos committed:
More signature updates in the MacOS related files

livecodepanos committed:
Updated some globals listed in the Unicodification spreadsheet

livecodesebastien committed:
dskmac.cpp purified - calls to MCS_* functions within MCMacDesktop changed to MCMacDesktop:: equivalent instead

livecodepanos committed:
Modifications after Mark's feedback #4

livecodepanos committed:
Updated some more globals

livecodepanos committed:
Updated MCfiletype global (it was incomplete in previous commit)

livecodesebastien committed:
All straightforward Mac desktop functions refactored. Functions common to several platforms are still to be refactored

livecodesebastien committed:
Refactoring of the Mac specific functions continued - some changes previously made have been reverted as some functions are to be removed and didn't need to be exposed

livecodepanos committed:
Updated all the static vars listed in the Unicodification spreadsheet

livecodesebastien committed:
Windows functions refactoring - commit from Mac, to have the modifications available from Windows

livecodeali committed:
Merge branch 'refactor-syntax' into refactor-syntax_unicode

livecodepanos committed:
Updated the MCcmd global (there are no more globals left to update)

livecodepanos committed:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode

livecodepanos committed:
Modifications after Mark's feedback #5

livecodepanos committed:
Some more improvements after Mark's feedback #5

livecodepanos committed:
Fixed the ssl cert stuff in opensslsocket.cpp

livecodesebastien committed:
Signature of MCS_read* functions updated to the more convenient (void*, size_t, IO_header)
MCMemoryFileHandle and MCCustomFileHandle declarations moved to system.h so that they are accessible from platform-specific files (dskmac.cpp, dskw32.cpp and such)

runrevmark committed:
Merge pull request #137 from runrevpanos/refactor-syntax_unicode

Refactor syntax unicode

livecodepanos committed:
Fixed conflicts for merging upstream/refactor-syntax_unicode into refactor-syntax_unicode_globals_2 branch

livecodepanos committed:
fixed conflicts and reverted back changes in EXEC LEGACY

runrevmark committed:
[[ RefactorSyntax ]] Tweaks to make Mac server build (Desktop awaiting MCS_* update).

runrevmark committed:
[[ Foundation ]] Added MCValueAssign utility method.

runrevmark committed:
Merge pull request #140 from runrevmark/refactor-syntax_unicode

[[ Foundation ]] Added MCValueAssign utility method.

runrevmark committed:
[[ RefactorSyntax ]] Added MCStringToInteger / MCStringToUInt16 / MCStringToInt16 / MCStringToDouble utility methods.

runrevmark committed:
Merge branch 'refactor-syntax_unicode' of https://github.com/runrev/livecode into refactor-syntax_unicode

livecodepanos committed:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_globals_2

runrevmark committed:
[[ RefactorSyntax ]] Tweak to MCString primitives just added (they didn't compile).

runrevmark committed:
[[ UnicodifyScrollbar ]] Converted startstring / endstring of MCScrollbar to MCStringRef.

livecodepanos committed:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_globals_2

livecodesebastien committed:
MCS_init process reverted to its former execution; the refactoring was introducing problems with Windows

livecodesebastien committed:
Common initialization adapted for server and mobile implementations

livecodepanos committed:
Modifications on globals after Mark's feedback #6

Commit:
Add Unicode support to MCGraphic

MCGraphic now uses an MCStringRef to store its label. Wrapper functions MCFontMeasureText and MCFontDrawText were added - these take an MCStringRef rather than a {const char * p_text, size_t p_text_length, bool is_unicode) triple. The wrappers will need modification when MCScreenDC can take an MCStringRef in its corresponding methods.

Commit:
Add Unicode support to MCField

MCField now uses an MCStringRef to store its label.

livecodepanos committed:
Some more modifications on globals after Mark's feedback #6

Commit:
Simplify MCField label handling

As per Mark's suggestions, simplified the label handling for MCField (the label should now never be set to nil, only the empty string).

livecodepanos committed:
Did the Player Object fields

livecodepanos committed:
Some more updates on globals..

runrevmark committed:
Merge pull request #139 from runrevpanos/refactor-syntax_unicode_globals_2

Refactor syntax unicode globals 2

livecodepanos committed:
Updated fields MCStackfile.stackname and MCStackfile.filename of Stack Object

livecodepanos committed:
fixed conflicts so as to merge upstream/refactor-syntax_unicode into .._statics_1

Commit:
Further cleanup of MCGraphic

The getters/setters for the label property of MCGraphic have been cleaned up. The {Get,GetEffective,Set}UnicodeLabel functions now work with MCDataRef binary strings instead of MCStringRef. Do{Get,Set}Label have been removed.

The MCFont{Measure,Draw} wrappers have been fixed to work properly with StringRefs in both native and UTF-16 encodings.

All label manipulation functions now reject nil StringRefs - the no-label case is represented by setting the label to kMCEmptyString instead of nil.

MCGraphic::getlabeltext() now has get semantics, returning the label without a refcount bump.

An overload of IO_write_stringref has been added with a parameter explicitly requesting the encoding to use when writing out the string, regardless of the underlying encoding used by the given StringRef. A corresponding IO_read_stringref has also been added.

livecodepanos committed:
Updated some statics and globals

Commit:
Allow MCStringConvertToUTF8 to return a length

The existing MCStringConvertToUTF8 function calculated the byte length of a UTF-8 encoded string internally but did not provide a mechanism for returning this. A wrapper has been added for existing callers with no interest in the length.

Commit:
Add IO_{read,write}_stringref_utf8

These functions read a UTF-8 encoded string into a StringRef and write a StringRef as a UTF-8 encoded byte stream, respectively.

Commit:
Modified MCTooltip and MCObject to use StringRef tooltips

The tooltips used and stored by MCObject and MCTooltip now use StringRefs and support Unicode for I/O.

livecodesebastien committed:
index on refactor-syntax_unicode: f8fc9c0 Common initialization adapted for server and mobile implementations

livecodesebastien committed:
WIP on refactor-syntax_unicode: f8fc9c0 Common initialization adapted for server and mobile implementations

Commit:
Change MCControl::{Get,Set}UnicodeToolTip to take DataRef

These methods now take a DataRef containing UTF-16 encoded characters when explicitly requesting Unicode.

Commit:
Rename new MCStringConvertToUTF8String to MCStringConvertToUTF8

Function renamed in order to better reflect its operation. MCStringConvertToUTF8String is implemented in terms of the renamed function.

runrevmark committed:
[[ Foundation ]] Added declaration for MCStringEncode / MCStringEncodeAndRelease / MCStringDecode / MCStringDecodeAndRelease
[[ Foundation ]] Added declaration for MCStringCreateWithBytesAndRelease / MCStringCreateWithCharsAndRelease / MCStringCreateWithWStringAndRelease / MCStringCreateWithCFStringAndRelease.

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_graphic

Requires updated libfoundation/foundation.h header from upstream in order to clean up some StringRef/DataRef conversion code.

Commit:
Simplify the new IO_{read,write}stringref functions

The implementation of these methods is much simpler, making use of the MCStringEncode and MCStringCreateWithBytesAndRelease functions.

Commit:
Simplify MCGraphic::{Get,Set}UnicodeLabel

These methods now use the new MCString{Encode,Decode}{,AndRelease} functions, making their implementations much simpler.

Commit:
Merge remote-tracking branch 'upstream/refactor-syntax_unicode' into refactor-syntax_unicode_object

The new string manipulation methods in libfoundation/foundation.h are required in order to simplify some conversion code.

Commit:
Fix to IO_read_stringref_utf8

Updated the function to remove a resource leak.

Commit:
Simplify MCControl::{Get,Set}UnicodeToolTip

These methods now use the MCString{Encode,Decode} functions in order to simplify their implementation.

livecodesebastien committed:
Merge branch 'refactor-syntax' into refactor-syntax_unicode

livecodesebastien committed:
Merge branch 'temp-branch' into refactor-syntax_unicode

livecodepanos committed:
Updates for Stack Object fields and methods

livecodesebastien committed:
Fixed filehandling and string splitting bugs. The IDE now can start

livecodesebastien committed:
Implicit use of copy operator rendered impossible for MCAuto* classes

Commit:
Simplify MCGraphic::SetLabel

SetLabel was more complicated that necessary; the input parameter can never be nil.

Commit:
Removed un-necessary checks for nil

runrevmark committed:
Merge pull request #153 from runrevfraser/refactor-syntax_unicode_field_clean

Refactor syntax unicode field clean

Commit:
Further cleanup

Cleaning up some minor bits and pieces that got missed previously.

runrevmark committed:
Merge pull request #150 from runrevfraser/refactor-syntax_unicode_graphic

Refactor syntax unicode graphic

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_object

livecodepanos committed:
Updates after feedback on statics

Commit:
Rename MCField::settext to settext_oldstring

The MCString variant of MCField::settext has been renamed to MCField::settext_oldstring. In its place, the new MCField::settext takes a StringRef.

livecodesebastien committed:
Fixed utf8 conversion bug - no null-terminating character
MCS_pathtoref_and_leaf: condition corrected

runrevmark committed:
Merge pull request #155 from runrevfraser/refactor-syntax_unicode_field_clean

Rename MCField::settext to settext_oldstring

livecodesebastien committed:
License activation bug fixed

Commit:
Cleaning and adding new method MCTooltip::cleartip

The new method should be used whenever a tooltip message is to be set to empty. It is no longer permitted to call settip(NULL) (though settip(kMCEmptyString) will still work).

Additional minor clean-up has been performed.

Commit:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_object

Merge with upstream branch before pull into upstream.

livecodepanos committed:
Updates after feedback on statics plus updates in srvcgi.cpp

livecodepanos committed:
Updates after feedback on Player Object #1

livecodesebastien committed:
Merged runrev/refactor-syntax_unicode into runrevsebastien/refactor-syntax_unicode_mac

livecodepanos committed:
reverted changes in the MCImage obj var files

livecodepanos committed:
Updates on statics (removed the srv* files updates)

livecodepanos committed:
Added MCAndroidCalendarInitialize() and MCAndroidCalendarFinalize() methods

livecodepanos committed:
Updates on Image object field aand methods

runrevmark committed:
Merge pull request #162 from runrev/refactor-syntax_unicode_scrollbar

[[ UnicodifyScrollbar ]] Converted startstring / endstring of MCScrollba...

livecodepanos committed:
Updates on MCAndroidPurchace Object

livecodepanos committed:
Fixed a typo in previous commit

Commit:
Change MCControl::SetToolTip logic

Simplified the logic for tooltip updates.

livecodepanos committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_statics_1

livecodepanos committed:
Reverting server files to beginning of branch.

livecodesebastien committed:
Fixed tagging for the MCS_ functions

runrevmark committed:
Merge pull request #147 from runrevpanos/refactor-syntax_unicode_Player

Refactor syntax unicode player

runrevmark commented:
Hi Panos - could you merge in upstream refactor-syntax_unicode into this branch, then I'll review - thanks!

livecodepanos committed:
Final tweaks done, now is ready to merge

runrevmark committed:
Merge pull request #146 from runrevpanos/refactor-syntax_unicode_statics_1

Refactor syntax unicode statics 1

livecodepanos committed:
Some more updates

Commit:
Simplify MCTooltip::settip logic

MCTooltip::settip now uses cleartip() internally when attempting to set the tooltip to the empty string.

runrevmark committed:
Merge pull request #154 from runrevfraser/refactor-syntax_unicode_object

Refactor syntax unicode object

livecodepanos committed:
Fixed conflicts for merging upstream/refactor-syntax_unicode into refactor-syntax_unicode_Stack

livecodesebastien committed:
Tweaks to make things compile.

Commit:
Convert MCAd to use a StringRef name

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

Conflicts:
	engine/src/opensslsocket.cpp
	engine/src/osxspec.cpp

livecodesebastien committed:
Dummy implementations of MCStringEncode/Decode/CreateWithBytes

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

Conflicts:
	libfoundation/include/foundation.h

livecodesebastien committed:
Initial implementation of new MCString methods.

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodesebastien committed:
Implemented MCStringCreateWithCharsAndRelease

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodesebastien committed:
[[ RefactorSyntax ]] Corrected tooltip property type.
[[ RefactorSyntax ]] Fixed UTF8 encode / decode.

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

Conflicts:
	engine/src/dskmain.cpp

livecodesebastien committed:
[[ UnicodifyGraphic ]] Changed assign to retain in copy constructor.
[[ UnicodifyTooltip ]] Ensured array index lookups start at 1.

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodepanos committed:
fixed conflict to merge upstream/refactor-syntax_unicode into this branch

livecodepanos committed:
Merge branch 'refactor-syntax_unicode' of https://github.com/runrev/livecode into refactor-syntax_unicode_Image

livecodesebastien committed:
Make sure use of unicode when using IO_read/write_stringref is explicit.
Added check for nil string when using IO_read_stringref_utf8.

livecodesebastien committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodepanos committed:
Updates after Mark's feedback #1

livecodesebastien committed:
[[ refactorSyntax ]] Readall / readfixed calls sorted out
[[ refactorSyntax ]] ReadAvailable added to MCSystemFileHandle

Commit:
Minor cleanups of MCAd

livecodepanos committed:
Added MCJavaStringToStringRef method and used it

Commit:
Remove un-necessary checks for empty strings

An empty ad name is permissable and should be treated like any other ad.

runrevmark committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

Conflicts:
	engine/src/dskmain.cpp

livecodesebastien committed:
MCSystemFileHandle -> Read simplified to a unique function.
MCS_readall and MCS_readfixed wrap this function

runrevmark committed:
Make sure we clear globals on desktop.

livecodesebastien committed:
IO_header suppressed, MCSystemFileHandle now directly used.
Note: legacy code using IO_WRITTEN/IO_SEEKED flags has been commented out using #ifdef OLD_IO_HANDLE macro

livecodesebastien committed:
Merge remote-tracking branch 'origin/refactor-syntax_unicode_mac' into refactor-syntax_unicode_mac

Conflicts:
	engine/src/dskmain.cpp

livecodepanos committed:
Updated some methods of MCPlayer obj

runrevmark committed:
Removed a couple of unnecessary MCValueReleases (since MCArrayFetchValueAtIndex is a 'get' not a 'copy').

runrevmark committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodesebastien committed:
Merge remote-tracking branch 'origin/refactor-syntax_unicode_mac' into refactor-syntax_unicode_mac

runrevmark committed:
Merge pull request #166 from runrevfraser/refactor-syntax_unicode_ad

Refactor syntax unicode ad

livecodesebastien committed:
MCSerialPortFileHandle added for Mac - Read and Write might need to be updated though - see #if 0'd alternate implementation

livecodesebastien committed:
MCstacknames error on Livecode IDE closing fixed, as well as a memory leak.

livecodepanos committed:
Improvements in MCJavaStringToStringRef method

livecodepanos committed:
fixed conflicts to merge upstream/refactor-syntax_unicode into this branch

Commit:
StringRef-ify MCSensorPostErrorMessage

Android needs further work.

livecodesebastien committed:
MCSerialPortFileHandle: forgotten functions implemented
MCStdioFileHandle: m_is_eof was assigned Boolean instead of bool, fixed

runrevmark committed:
Reverting changes to server projects.

runrevmark committed:
Merge branch 'runrevpanos-refactor-syntax_unicode_Image' into refactor-syntax_unicode

runrevmark committed:
[[ UnicodifyImage ]] Make sure we force write_stringref to output native.

runrevmark commented:
Hi Panos - sorry, you need to merge in upstream again :)

runrevmark committed:
[[ UnicodifyPlayer ]] Make sure we write out filename stringref as native.

livecodepanos committed:
Fixed conflicts to merge upstream/refactor-syntax_unicode into this branch

Commit:
Simplified error message handling

livecodepanos committed:
Fixed some errors after the merge. It has still some in libfoundation/src/foundation-string.cpp

runrevmark committed:
Merge pull request #171 from runrevfraser/refactor-syntax_unicode_mblsensor

Refactor syntax unicode mblsensor

livecodepanos committed:
Updates after feedback #1 on this branch

runrevmark commented on engine/src/stack3.cpp:
Here it should actually be 'false' - the third parameter specifies whether to write out the contents of the StringRef as unicode or not. In almost all cases, we don't want to write out as unicode (so parameter should be false) as the file-format doesn't like Unicode strings anywhere except specific places at the moment.

runrevmark commented on engine/src/stack3.cpp:
Should be 'false' here.

runrevmark commented on engine/src/stack3.cpp:
This should be IO_write_stringref_utf8

livecodepanos committed:
Removed some (no longer used) commented code

runrevmark committed:
Merge pull request #169 from runrevpanos/refactor-syntax_unicode_MCPlayer_methods

Refactor syntax unicode mc player methods

runrevmark committed:
Merge pull request #163 from runrevpanos/refactor-syntax_unicode_MCAndroidPurchace

Refactor syntax unicode mc android purchace

livecodepanos committed:
Updates after feedback #1 on this branch

Commit:
Update MCSystemLaunchUrl on iOS

The Android version already took a StringRef but the iOS version had not been updated.

runrevmark committed:
Merge pull request #176 from runrevfraser/refactor-syntax_unicode_mcsystemlaunchurl

Refactor syntax unicode mcsystemlaunchurl

livecodesebastien committed:
- MCS_umask removed, replaced where is was used with MCS_getumask or MCS_setumask
- legacy functions definitions from osxprefix.h moved to a new file osxprefix-legacy.h
- MCS_isatty replaced with MCS_isinteractiveconsole; isInteractiveConsole added in MCSystemInterface.
- MCS_handle_socket made into a platform independent functions
- Declaration of MCS functions defined in opensslsocket.cpp moved to osspec.h
- MCS_write_socket: use of strchr with const char* fixed.
- Mac specific MCS functions renamed to MCS_mac and it applied, moved from osspec.cpp/osxfile.cpp to dskmac.cpp.
- <void MCS_pathtoref(const char*, FSRef)> totally replaced with <bool MCS_pathtoref(MCStringRef, FSRef)> and removed
- <char* MCS_fsref_to_path(FSRef)> replaced with <bool MCS_fsref(FSRef, MCStringRef&)> and removed
- MCS_imeisunicode removed
- MCS_nodelay made private to Mac - and Linux
- MCS_isnan made platform independent and implemented in sysspec.h
- MCSystemInterface::OpenStdFile renamed OpenFd
- MCSystemInterface::ResolveNativePath removed as it is the same as ResolvePath

runrevmark committed:
[[ RefactorSyntax ]] Corrections to array saving and loading code.

runrevmark committed:
Merge branch 'refactor-syntax' into refactor-syntax_unicode

Conflicts:
	engine/src/control.cpp

runrevmark committed:
Merge branch 'refactor-syntax_unicode' into refactor-syntax_unicode_mac

livecodesebastien committed:
- <char *MCS_mac_FFSpec2path(FFSpec*)> changed to <bool MCS_mac_FSSpec2path(FSSpec *, MCStringRef&)> - and all cascading changes implied
- MCS_get_canonical_path removed
- strchr with const char* fixed in <OSerr getDescFromAddress()>
- MCU_path2native and MCU_path2std functions replaced with MCS_path(from|to)native

runrevmark committed:
[[ UnicodifyImage ]] Fixed error with filename setting.
[[ UnicodifyPlayer ]] Fixed usage of filename field.

livecodesebastien committed:
Merge remote-tracking branch 'origin/refactor-syntax_unicode_mac' into refactor-syntax_unicode_mac

livecodesebastien committed:
Merge pull request #185 from runrevsebastien/refactor-syntax_unicode_mac

Refactor syntax unicode mac

runrevmark committed:
[[ UnicodifyMac ]] Removed moved code from osxspec.cpp and osxfiles.cpp.

runrevmark committed:
[[ UnicodifyMac ]] Removed unused files (will be reinstated when we next merge in master).

runrevmark committed:
[[ UnicodifyImage ]] MCImage::F_HAS_FILENAME now only relevant in save/load.
[[ UnicodifyImage ]] Editability of image now determined by if filename is empty.
[[ Unicodify ]] Fixed error in answer file with types - initial is optional (so should be nil if not specified).

runrevmark committed:
Merge branch 'refactor-syntax' into refactor-syntax_unicode

runrevmark committed:
[[ UnicodifyPlayer ]] Change Assign to Retain in MCPlayer copy constructor.

runrevmark committed:
Merge branch 'refactor-syntax' into refactor-syntax_unicode

runrevmark committed:
[[ Unicodify ]] Corrected MCU_puturl - previously it was using the wrong filename (not trimmed).

livecodepanos committed:
Merged in runrev/refactor-syntax_unicode into this branch. It compiles on Mac desktop (builds successfully and a window asking for login details pops up, but after this I am not sure what happens

runrevmark commented:
Hi Panos - I've merged this into runrev/refactor-syntax_unicode via runrev/refactor-syntax_unicode_stack.
